src/main/java/seedu/address/storage/JsonAdaptedRecord.java

```diff
@@ -0,0 +1,72 @@
+package seedu.address.storage;
+
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+import seedu.address.commons.exceptions.IllegalValueException;
+import seedu.address.model.record.Condition;
+import seedu.address.model.record.DateTime;
+import seedu.address.model.record.Record;
+
+/**
+ * Jackson-friendly version of {@link Record}.
+ */
+public class JsonAdaptedRecord {
+
+    public static final DateTimeFormatter FORMATTER = DateTimeFormatter.ofPattern("ddMMyyyy HHmm");
+    public static final String MISSING_FIELD_MESSAGE_FORMAT = "Person's %s field is missing!";
+    private final LocalDateTime dateTime;
+    private final List<JsonAdaptedCondition> conditions = new ArrayList<>();
+
+
+    /**
+     * Constructs a {@code JsonAdoptedRecord} with the given record details.
+     */
+    @JsonCreator
+    public JsonAdaptedRecord(@JsonProperty("dateTime") LocalDateTime dateTime,
+                             @JsonProperty("condition") List<JsonAdaptedCondition> conditions) {
+        this.dateTime = dateTime;
+        if (conditions != null) {
+            this.conditions.addAll(conditions);
+        }
+    }
+
+    /**
+     * Converts a given {@code Record} into this class for Jackson use.
+     */
+    public JsonAdaptedRecord(Record source) {
+        this.dateTime = source.getDateTime().dateTime;
+        this.conditions.addAll(source.getConditions().stream()
+                .map(JsonAdaptedCondition::new)
+                .collect(Collectors.toList()));
+    }
+
+    /**
+     * Converts this Jackson-friendly adapted person object into the model's {@code Person} object.
+     *
+     * @throws IllegalValueException if there were any data constraints violated in the adapted person.
+     */
+    public Record toModelType() throws IllegalValueException {
+        final List<Condition> conditionsList = new ArrayList<>();
+        for (JsonAdaptedCondition condition : conditions) {
+            conditionsList.add(condition.toModelType());
+        }
```

Comment:
This is where the condition is not null check should be done, in the case where the JSON Record is messsed up and there is an empty condition for a Record, then should throw Exception

Reply:
Fixed.